WeakAurasOptions/AceGUI-Widgets/AceGUIWidget-WeakAurasDisplayButton.lua

```diff
@@ -1629,17 +1501,50 @@ local methods = {
           if not (self.data.skipWagoUpdate and self.data.skipWagoUpdate == updateData.wagoVersion) then
             if tonumber(updateData.wagoVersion) > tonumber(self.data.version) then
               -- got update
-              return true, false, updateData
+              return true, false, updateData, slug
             end
           else
             -- version skip flag
-            return true, true, updateData
+            return true, true, updateData, slug
           end
         end
       end
     end
     -- no addon, or no data, or ignore flag
-    return false, false, nil
+    return false, false, nil, nil
+  end,
+  ["RefreshUpdateIcon"] = function(self, actionFunc, skipRefreshGroup)
+    if self.data.parent then
+      -- is in a group
+      if not skipRefreshGroup then
+        local button = WeakAuras.GetDisplayButton(self.data.parent)
+        if button then
+          WeakAuras.RefreshGroupUpdateIcon(button)
+        end
+      end
+    else
+      if self.data.controlledChildren then
+        -- is a group
+        if actionFunc then
+          for childIndex, childId in pairs(self.data.controlledChildren) do
+            local button = WeakAuras.GetDisplayButton(childId)
+            if button then
+              button.callbacks[actionFunc](nil, true)
+            end
+          end
+        end
+        WeakAuras.RefreshGroupUpdateIcon(self)
+      else
+        local hasUpdate, skipVersion = self:HasUpdate()
```

Comment:
This code is almots identically to what RefreshGroupUpdateIcon does.

If I see that correctly, making RefreshGroupUpdateIcon handle top level auras in addition to groups requires just adding a check for data.controlledChildren around the loop.

And then this code can be simplified a bit too.

Otherwise, it took me a bit to figure out how the recursion works. I don't have a suggestion on how to write that clearer though.

Reply:
i removed WeakAuras.RefreshGroupUpdateIcon, it was indeed unnecessary and combining them should make the code a bit easier to understand